lib/alerts/check_timeline_alert_manager.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class CheckTimelineAlertManager
+  def initialize(course)
+    @course = course
+    create_alerts
+  end
+
+  def create_alerts
+    return unless @course.approved?
+    # if course doesn't have any training module, it will not create an alert
+    return if @course.training_module_ids.any?
+    return if alert_already_exists?(@course.id)
+    alert = Alert.create(type: 'CheckTimelineAlert',
+                         course_id: @course.id,
```

Comment:
It's slightly more elegant and Rails-y to do this with `course: @course` instead of `course_id: @course.id`. (They do the same thing.